python/ray/data/dataset.py

```diff
@@ -2619,6 +2625,21 @@ def write_datasource(
         finally:
             progress.close()
 
+    def iterator(self) -> DatasetIterator:
+        """Return a :class:`DatasetIterator <ray.data.DatasetIterator>` that
+        can be used to repeatedly iterate over the dataset.
+        Examples:
```

Comment:
Nit: Add blank line between docstring summary and examples.
```suggestion

        Examples:
```